packages/nodejs/src/typeTranslation.ts

```diff
@@ -235,6 +244,116 @@ function transBaker(
+        accountCreationLimit: unwrap(params.accountCreationLimit?.value),
+        foundationAccount: v1.AccountAddress.fromBytes(
+            Buffer.from(unwrap(params.foundationAccount?.value))
+        ).address,
+    };
+}
+
+function translateCommissionRange(
+    range: v2.InclusiveRangeAmountFraction | undefined
+): v1.InclusiveRange<number> {
+    return {
+        min: transAmountFraction(range?.min),
+        max: transAmountFraction(range?.max),
+    };
+}
+
+function translateRewardParametersCommon(
+    params: v2.ChainParametersV1 | v2.ChainParametersV0
+): v1.RewardParametersCommon {
+    return {
+        transactionFeeDistribution: {
+            baker: transAmountFraction(
+                params.transactionFeeDistribution?.baker
+            ),
+            gasAccount: transAmountFraction(
+                params.transactionFeeDistribution?.gasAccount
+            ),
+        },
+        gASRewards: {
+            baker: transAmountFraction(params.gasRewards?.baker),
+            finalizationProof: transAmountFraction(
+                params.gasRewards?.finalizationProof
+            ),
+            accountCreation: transAmountFraction(
+                params.gasRewards?.accountCreation
+            ),
+            chainUpdate: transAmountFraction(params.gasRewards?.chainUpdate),
+        },
+    };
+}
+
+function translateMintRate(mintRate: v2.MintRate | undefined): number {
+    return unwrap(mintRate?.mantissa) * 10 ** (-1 * unwrap(mintRate?.exponent));
+}
+
+function transPoolPendingChange(
+    change: v2.PoolPendingChange | undefined
+): v1.BakerPoolPendingChange {
+    switch (change?.change?.oneofKind) {
+        case 'reduce': {
+            return {
+                pendingChangeType:
+                    BakerPoolPendingChangeType.ReduceBakerCapital,
+                effectiveTime: new Date(
+                    Number(unwrap(change.change.reduce.effectiveTime?.value))
+                ),
+                bakerEquityCapital: unwrap(
+                    change.change.reduce.reducedEquityCapital?.value
+                ),
+            };
+        }
+        case 'remove': {
+            return {
+                pendingChangeType: BakerPoolPendingChangeType.RemovePool,
+                effectiveTime: new Date(
+                    Number(unwrap(change.change.remove.effectiveTime?.value))
+                ),
+            };
+        }
+        default:
+            return {
+                pendingChangeType: BakerPoolPendingChangeType.NoChange,
+            };
+    }
+}
+
+function transPoolInfo(info: v2.BakerPoolInfo): v1.BakerPoolInfo {
+    return {
+        openStatus: transOpenStatus(info.openStatus),
+        metadataUrl: info.url,
+        commissionRates: transCommissionRates(info.commissionRates),
+    };
+}
+
+function transPaydayStatus(
+    status: v2.PoolCurrentPaydayInfo | undefined
+): v1.CurrentPaydayBakerPoolStatus | undefined {
+    if (!status) {
+        return undefined;
```

Comment:
Is there a reason for not just unwrapping status? Can it be undefined in v2?

Reply:
Yeah:
```
* @generated from protobuf field: optional concordium.v2.PoolCurrentPaydayInfo current_payday_info = 8;
```